src/components/Modals/FlexibleModal.vue

```diff
@@ -0,0 +1,117 @@
+<template>
+  <div class="modal">
+    <div v-bind:class="['modal-content', contentClass]">
+      <div class="modal-top">
+        <span class="modal-title">{{ title }}</span>
+        <img class="modal-exit" src="@/assets/images/x.png" @click="closeCurrentModal" />
+      </div>
+      <slot class="modal-body"></slot>
+      <div class="modal-buttonWrapper">
+        <button class="modal-button" @click="leftButtonClicked">{{ leftButtonText }}</button>
+        <button
+          class="modal-button modal-button--add"
+          :class="{ 'modal-button--disabled': rightButtonIsDisabled }"
+          @click="rightButtonClicked"
+        >
+          {{ rightButtonText }}
+        </button>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+
+export default Vue.extend({
+  props: {
+    title: { type: String, required: true },
+    contentClass: { type: String, required: true },
+    leftButtonText: { type: String, required: true },
+    rightButtonText: { type: String, required: true },
+    rightButtonIsDisabled: { type: Boolean, required: true },
+  },
+  methods: {
+    closeCurrentModal(): void {
+      this.$emit('modal-closed');
+    },
+    leftButtonClicked(): void {
+      this.$emit('left-button-clicked');
+    },
+    rightButtonClicked(): void {
+      this.$emit('right-button-clicked');
+    },
+  },
+});
+</script>
+
+<style lang="scss">
+@import '@/assets/scss/_variables.scss';
+
+.modal {
+  padding: 1rem;
+
+  &-content {
+    background: $white;
+    border-radius: 9px;
+    margin-left: auto;
+    margin-right: auto;
+    padding: 1rem;
+  }
+
+  &-body {
+    padding: 0;
+  }
+
+  &-top {
+    display: flex;
+    justify-content: space-between;
+    margin-bottom: 0.5rem;
+  }
+
+  &-exit {
+    cursor: pointer;
+    width: 10.5px;
+    height: 10.5px;
+  }
+
+  &-title {
+    font-weight: bold;
+    font-size: 16px;
+    line-height: 20px;
+    color: #3d3d3d;
+  }
+
+  &-buttonWrapper {
+    margin-top: 1rem;
+    display: flex;
+    justify-content: flex-end;
+  }
+
+  &-button {
+    width: 4.75rem;
+    height: 1.8rem;
+    color: #5b676d;
+    border-radius: 3px;
+    border: 1px solid #3d3d3d;
```

Comment:
Not sure if this is out of scope of this PR, but we should change the color values to the scss variables. I believe #3d3d3d is $primaryGray. Some of the color values like #5b676d do not seem to have variables so maybe we want to add them.

Reply:
changed